### ISSUE TITLE
A question about the SOS and FOS

### ISSUE THREAD

**Issue by huanghanmei** [tonyngjichun/SOLAR#1]
I am reading you paper but i can not understand the difference between SOS and FOS.
By the way ,can you release you training code?
thank you!

**Comment by tonyngjichun** [comment]
Hi

SOS acts as a regularisation term on top of the triplet loss (which we refer to as the FOS in the paper) - intuitively from Eq. (7) it minimises the _difference_ in distance **between anchor & negatives** and **between positive & negative**s, which encourages more compact clustering and using more available area on the unit hypersphere. If you could refer to SOSNet: https://arxiv.org/abs/1904.05019 to understand a bit more about the principles and experiments behind it. In our paper, the SOS loss differs slightly in notation, as there are no explicit negatives in SOSNet's settings and we do not perform hard-negatives mining within the batch but precompute from a pool of candidate negatives at the start of each epoch, but the mechanism is identical to that in SOSNet.

As for the training code, we are planning to release it during/right after the ECCV week (23-28 Aug).

Best,
Tony Ng

**Comment by huanghanmei** [comment]
Thanks for your answer !
